### PR TITLE
feat(UI-914): remove 'warning active deployment' modal from edit forms

### PR DESCRIPTION
**Pull request by RonenMars**
## Description
When we have active deployment, and we are adding/editing connection/trigger/variable we have a warning dialog.
Display the dialog before we click add/edit on the table page.

## Linear Ticket
https://linear.app/autokitteh/issue/UI-914/this-might-affect-active-deployment-before-save-on-editadd-click

## What type of PR is this? (check all applicable)

-   [x] 💡 (feat) - A new feature (non-breaking change which adds functionality)
-   [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
-   [ ] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
-   [ ] 🏎 (perf) - Optimization
-   [ ] 📄 (docs) - Documentation - Documentation only changes
-   [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
-   [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] ☑️ (chore) - Chores - Other changes that don't modify src or test files
-   [ ] ↩️ (revert) - Reverts - Reverts a previous commit(s).

### CODE REVIEW

File: src/components/organisms/activeDeploymentWarningModal.tsx [hunk 1]
```diff
@@ -3,16 +3,25 @@ import React from "react";
...
+	goToAdd,
+	goToEdit,
+	modifiedId,
+}: ActiveDeploymentWarningModalProps) => {
 	const { t } = useTranslation("modals", { keyPrefix: "warningActiveDeployment" });
 	const { closeModal } = useModalStore();
 
+	if (!action) return null;
+
+	const onOkClick = () => (action === "edit" ? goToEdit(modifiedId) : goToAdd());
```
[the remark this answers is not in the document]
Comment (RonenMars) [reply src/components/organisms/activeDeploymentWarningModal.tsx:23]: Why not :)?

File: src/components/organisms/connections/table.tsx [hunk 2]
```diff
@@ -26,16 +27,44 @@ export const ConnectionsTable = () => {
...
 	const {
 		connections,
 		fetchConnections,
+		hasActiveDeployments,
 		loading: { connections: isLoading },
 	} = useCacheStore();
 	const { items: sortedConnections, requestSort, sortConfig } = useSort<Connection>(connections || [], "name");
 	const { resetChecker, setFetchConnectionsCallback } = useConnectionCheckerStore();
 
+	// Add state for warning modal action
```
[the remark this answers is not in the document]
Comment (RonenMars) [reply src/components/organisms/connections/table.tsx:40]: removed

File: src/components/organisms/triggers/table.tsx [hunk 3]
```diff
@@ -3,6 +3,7 @@ import React, { useCallback, useMemo, useState } from "react";
 import { useTranslation } from "react-i18next";
 import { useNavigate, useParams } from "react-router-dom";
 
+import { ActiveDeploymentWarningModal } from "../activeDeploymentWarningModal";
```
[the remark this answers is not in the document]
Comment (RonenMars) [reply src/components/organisms/triggers/table.tsx]: corrected

**Review by J1za — APPROVED** [review]

File: src/components/organisms/activeDeploymentWarningModal.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (J1za) [reply src/components/organisms/activeDeploymentWarningModal.tsx:23]: I don't like the name. ;)

[the remark this answers is not in the document]
Comment (RonenMars) [reply src/components/organisms/activeDeploymentWarningModal.tsx:23]: Any other suggestions?

File: src/store/cache/useCacheStore.ts [hunk 4]
```diff
@@ -465,4 +464,10 @@ const store: StateCreator<CacheStore> = (set, get) => ({
 	},
 });
 
+const selectHasActiveDeployments = createSelector(
+	(state: CacheStore) => state.deployments,
+	(deployments) => deployments?.some((dep) => dep.state === DeploymentStateVariant.active) ?? false
```
Comment (J1za) [inline src/store/cache/useCacheStore.ts:469]:
```suggestion
	(deployments) => deployments?.some(({state}) => state === DeploymentStateVariant.active) || false
```

Comment (RonenMars) [reply src/store/cache/useCacheStore.ts:469]: done

**Review by J1za — APPROVED** [review]

**Pull request merged**